draft-setty-cfrg-spartan-incubation.md

```diff
@@ -33,15 +33,37 @@ informative:
+
+## zkSNARK Signatures
+
+We describe how to add selective disclosure and unlinkability to verifiable credentials issued with standard digital signatures over standardized curves. When a credential holder wishes to provably disclose a subset of attribute-value pairs in their credential to a verifier, the credential holder produces a derived credential containing only the disclosed attribute-value pairs. Such derived credentials are then presented to the verifier who can verify the validity of the credential.
+
+A challenging aspect here is: how does a credential holder produce such derived credentials, with selective disclosure and unlinkability guarantees? Note that the credential holders cannot “call the issuer” for any interaction with verifiers, as this poses additional privacy threats. Furthermore, credential holders should not be able to derive fake credentials. Standard signature schemes such as ECDSA or EdDSA do not support such functionality.
+
+We address both requirements with zkSNARKs, a cryptographic primitive that allows a prover to prove an arbitrary mathematical statement to a verifier. In particular, we plan to leverage Spartan, a high-speed zkSNARK that uses only standardized cryptography and offers state-of-the-art performance. Spartan also does not require any trusted setup (e.g., creating a structured reference string).
+
+An attractive aspect of our approach is that we can retrofit the aforementioned privacy properties to existing verifiable credentials issued with standard signature schemes (e.g., ECDSA over secp256k1 or P-256), with minimal resource overheads and engineering effort. Specifically, an issuer issues credentials as before with a standard signature scheme (e.g., ECDSA) so it is unmodified and incurs no overheads to support privacy features. Deriving a derived credential and verification of a derived credential happens on the credential holder and the verifier side, respectively. Specifically, the credential presentation protocol flow is as follows:
+* The issuer digitally signs a set of attribute-value pairs using its private key and sends ( to the holder, where is a digital signature (e.g., ECDSA signature).
```

Comment:
A typo? "sender ("

Reply:
This is your original text ;) IMO let's just merge it as-is and then make an editorial pass?

Reply:
The original text had symbols. Perhaps they were lost when turned into markdown?